core/resource/opened.go

```diff
@@ -8,23 +8,17 @@ import (
 	"io"
 
 	"github.com/juju/errors"
+
+	charmresource "github.com/juju/juju/internal/charm/resource"
 )
 
 // Opened provides both the resource info and content.
 type Opened struct {
-	Resource
+	Size        int64
+	Fingerprint charmresource.Fingerprint
```

Comment:
I think it would be useful to check the commit/PR which introduces it, and if there is nothing against it, just remove Opened type and use `Content` whenever we used a `Opened`

Reply:
Looks like any intent behind `Content` is going to be very hard to find. All the reviews are on that defunct vapour platform. It sounds like a good idea to get rid of the `Opener` type and just use that, will do that.

@hmlanigan looks like that should just use the GenerateContent, we can change this when we touch the uniter facade.

Reply:
On second thoughts, I might leave this as is in this PR for now but with the content method, and leave it to be refactored when the uniter facade is touched if nobody has any objections to that. It looks like the file that Heather pointed out should use this, and it uses the content method.

Reply:
On third thoughts this is too much of a distraction, I'm just going to leave it as it is. There would be no changes to production code for it, but test code is a mess. The testing helper functions return one of these and all the tests all over the API server rely on that to just get a resource and an opener at the same time.